scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -124,60 +200,68 @@ class RunCoverage(
           .retrieveCoverageDataForTestTarget(testTarget.removeSuffix(".kt"))
       }
 
+      coverageReports.forEach { report ->
+        if (report.hasFailure()) {
+          return CoverageReport.newBuilder()
+            .setFailure(report.failure)
+            .build()
+        }
+      }
+
       val aggregatedCoverageReport = calculateAggregateCoverageReport(coverageReports)
-      val reporter = CoverageReporter(repoRoot, aggregatedCoverageReport, reportFormat)
-      val (computedCoverageRatio, reportText) = reporter.generateRichTextReport()
+      return aggregatedCoverageReport
+    }
+  }
 
-      File(reportOutputPath).apply {
-        parentFile?.mkdirs()
-        writeText(reportText)
+  private fun combineCoverageReports(coverageResultList: List<CoverageReport>):
+    CoverageReportContainer {
+      val containerBuilder = CoverageReportContainer.newBuilder()
+      coverageResultList.forEach { report ->
+        containerBuilder.addCoverageReport(report)
       }
+      return containerBuilder.build()
+    }
```

Comment:
```suggestion
  private fun combineCoverageReports(
    coverageResultList: List<CoverageReport>
  ): CoverageReportContainer {
    val containerBuilder = CoverageReportContainer.newBuilder()
    coverageResultList.forEach { report ->
      containerBuilder.addCoverageReport(report)
    }
    return containerBuilder.build()
  }
```

The correct format wrapping for this case.

Reply:
Should have realized when they appeared a bit unusual. Thanks for the suggestion. Updated it.